packages/smart-wallet/src/typeGuards.js

```diff
@@ -0,0 +1,66 @@
+// @ts-check
+
+import { M } from '@agoric/store';
+import {
+  InstanceHandleShape,
+  ProposalShape,
+} from '@agoric/zoe/src/typeGuards.js';
+
+export const shape = {
+  // smartWallet
+  StringCapData: {
+    body: M.string(),
+    slots: M.arrayOf(M.string()),
+  },
+
+  // invitations
+  ContractInvitationSpec: M.split(
+    {
+      source: 'contract',
+      instance: InstanceHandleShape,
+      publicInvitationMaker: M.string(),
+    },
+    M.partial({
+      invitationArgs: M.array(),
+    }),
+  ),
+  ContinuingInvitationSpec: M.split(
+    {
+      source: 'continuing',
+      previousOffer: M.number(),
+      invitationMakerName: M.string(),
+    },
+    M.partial({
+      invitationArgs: M.array(),
+    }),
+  ),
+  PurseInvitationSpec: {
+    source: 'purse',
+    instance: InstanceHandleShape,
+    description: M.string(),
+  },
+
+  // offers
+  OfferShape: M.split(
+    {
+      id: M.number(),
+      // TODO M.unknown() to defer validation
+      invitationSpec: M.any(),
+      // XXX ProposalShape requires all whereas TS types ProposalRecord does and Proposal is partial
+      proposal: M.partial(ProposalShape),
+    },
+    M.partial({ offerArgs: M.recordOf(M.string(), M.any()) }),
```

Comment:
Only `offerArgs` is allowed in the type, so should this be `M.opt({ offerArgs: M.recordOf(M.string(), M.any()) })`?

Reply:
Otherwise this permits additional properties that are not in the offerSpec type

Reply:
can `offerArgs` be present with an `undefined` value? That might require a slightly different guard

Reply:
I think you may be missing the `M.split`. The desired type is https://github.com/Agoric/agoric-sdk/blob/70a070446ed63e9fdd7b4c911f76c871c3dbee8e/packages/smart-wallet/src/offers.js#L8-L15

So the shape should be,
```
  OfferSpec: M.split(
    // these three are required
    {
      id: M.number(),
      // TODO M.unknown() to defer validation
      invitationSpec: M.any(),
      // XXX ProposalShape requires all whereas TS types ProposalRecord does and Proposal is partial
      proposal: M.partial(ProposalShape),
    },
    // because of the split we need a partial
    M.partial({ offerArgs: 
     // should not be optional because if there is a key then it should be a defined value
      M.recordOf(M.string(), M.any())
  }),
  ),
```

But… I think the TS was too strict and `offerArgs` is really `any`. So I'll change that in both.